src/middleware/trailing-slash/index.test.ts

```diff
@@ -0,0 +1,57 @@
+import { Hono } from '../../hono'
+import { trimTrailingSlash, appendTrailingSlash } from '.'
+
+describe('Resolve trailing slash', () => {
+  let app: Hono
+
+  it('Trim', async () => {
+    app = new Hono({ strict: true })
+
+    app.use('*', trimTrailingSlash())
+
+    app.get('/the/example/endpoint/without/trailing/slash', async (c) => {
+      return c.text('301 moved permanently')
+    })
+
+    let resp: Response
+
+    resp = await app.request('/')
+    expect(resp).not.toBeNull()
+    expect(resp.status).toBe(200)
+
+    resp = await app.request('/the/example/endpoint/without/trailing/slash/')
+    expect(resp).not.toBeNull()
+    expect(resp.status).toBe(301)
+
+    resp = await app.request('/the/example/endpoint/without/trailing/slash')
+    expect(resp).not.toBeNull()
+    expect(resp.status).toBe(301)
```

Comment:
I think the expected value is `200`. In my thoughts, if the request URL does not have a tailing slash and enabled `trimTrailingSlash()`, it should not redirect. Wha do you think?